lib/internal/crypto/hash.js

```diff
@@ -63,6 +66,22 @@ const LazyTransform = require('internal/streams/lazy_transform');
 const kState = Symbol('kState');
 const kFinalized = Symbol('kFinalized');
 
+/**
+ * @param {string} name
+ */
+function normalizeAlgorithm(name) {
+  return StringPrototypeReplace(StringPrototypeToLowerCase(name), '-', '');
+}
+
+let warned = false;
+const emitDeprecationWarning = getDeprecationWarningEmitter(
+  'DEP0198',
+  'Creating SHAKE128/256 digests without an explicit options.outputLength is deprecated.',
+  undefined,
+  false,
+  isPendingDeprecation,
+);
```

Comment:
nit: Can we move this to the `shouldEmitWarning` callback? If you apply the following diff, that would allow us to have a  cleaner implementation IMO
```diff
diff --git a/lib/internal/util.js b/lib/internal/util.js
index 180ca49b32..59fcf4698e 100644
--- a/lib/internal/util.js
+++ b/lib/internal/util.js
@@ -110,8 +110,8 @@ function getDeprecationWarningEmitter(
   shouldEmitWarning = () => true,
 ) {
   let warned = false;
-  return function() {
-    if (!warned && shouldEmitWarning()) {
+  return function(arg) {
+    if (!warned && shouldEmitWarning(arg)) {
       warned = true;
       if (code === 'ExperimentalWarning') {
         process.emitWarning(msg, code, deprecated);
```
```suggestion
const emitDeprecationWarning = getDeprecationWarningEmitter(
  'DEP0198',
  'Creating SHAKE128/256 digests without an explicit options.outputLength is deprecated.',
  undefined,
  false,
  (algo) => {
    if (!isPendingDeprecation()) return false;
    const normalized = normalizeAlgorithm(algorithm);
    return normalized === 'shake128' || normalized === 'shake256';
  },
);
```